config_test.go

```diff
@@ -48,27 +48,30 @@ func TestLoadExternalComponentsFromConfig(t *testing.T) {
 	}
 	defer cleanUpFunc()
 
-	var cfg config
-	cfg.Builders = packersdk.MapOfBuilder{}
-	cfg.PostProcessors = packersdk.MapOfPostProcessor{}
-	cfg.Provisioners = packersdk.MapOfProvisioner{}
+	cfg := config{
+		Plugins: &packer.PluginConfig{
+			Builders:       packer.MapOfBuilder{},
+			PostProcessors: packer.MapOfPostProcessor{},
+			Provisioners:   packer.MapOfProvisioner{},
+		},
+	}
 
 	if err := decodeConfig(strings.NewReader(packerConfigData), &cfg); err != nil {
 		t.Fatalf("error encountered decoding configuration: %v", err)
 	}
 
 	cfg.LoadExternalComponentsFromConfig()
 
-	if len(cfg.Builders) != 1 || !cfg.Builders.Has("cloud-xyz") {
-		t.Errorf("failed to load external builders; got %v as the resulting config", cfg.Builders)
+	if len(cfg.Plugins.Builders) != 1 || !cfg.Plugins.Builders.Has("cloud-xyz") {
+		t.Errorf("failed to load external builders; got %v as the resulting config", cfg.Plugins.Builders)
 	}
 
-	if len(cfg.PostProcessors) != 1 || !cfg.PostProcessors.Has("noop") {
-		t.Errorf("failed to load external post-processors; got %v as the resulting config", cfg.PostProcessors)
+	if len(cfg.Plugins.PostProcessors) != 1 || !cfg.Plugins.PostProcessors.Has("noop") {
+		t.Errorf("failed to load external post-processors; got %v as the resulting config", cfg.Plugins.PostProcessors)
 	}
 
-	if len(cfg.Provisioners) != 1 || !cfg.Provisioners.Has("super-shell") {
-		t.Errorf("failed to load external provisioners; got %v as the resulting config", cfg.Provisioners)
+	if len(cfg.Plugins.Provisioners) != 1 || !cfg.Plugins.Provisioners.Has("super-shell") {
```

Comment:
I like that you've moved these into a plugins substruct -- makes it easier to mentally distinguish the template config from the plugins themselves.